File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        //edge cases, where one of the lists is empty
        if (list1 == nullptr && list2 == nullptr){
            return NULL;}
        if (list1 == nullptr){
            return list2;}
        if (list2 == nullptr){
            return list1;}
        ListNode* original = new ListNode(); //original node to store head.
        ListNode* curr = original; // curr to traverse lists
        while (list1 != NULL && list2 != NULL) {
            if (list1->val >= list2->val){// running through example, list.1 = list2.1, so next = list2.1. so now curr = 1->1, list2 is now 3->4.
                curr->next = list2;
                list2 = list2->next;
            }
            else{ // more example, gets ran next. so 1->1->
                curr->next = list1;
                list1 = list1->next;}
            curr = curr->next; //increment for next iteration. 
        }
        
        if (list1 == NULL) {
            curr->next = list2;
        }
        else {
            curr->next = list1;
        }
        
        return original->next;
            
            
            
    }
            
                
            
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        if (list1 == None):
            if (list2 == None):
                return 
            #only list 1 empty:
            else: 
                return list2
        
        temp_arr = []
        while(list1 != None and list2 != None):
            if (list1.val >= list2.val):
                temp_arr.append(list2.val)
                list2 = list2.next
            else: #(list2.val > list1.val)
                temp_arr.append(list1.val)
                list1 = list1.next
        try:
            if (list1.val != None):
                while(list1 != None):
                    temp_arr.append(list1.val)
                    list1 = list1.next
        except:
            pass
        try:
            if (list2.val != None):
                while(list2 != None):
                    temp_arr.append(list2.val)
                    list2 = list2.next
        except:
            pass
        ans = ListNode(temp_arr[0])
        ans_head = ans
        for i in range(1,len(temp_arr),1):
            ans.next = ListNode(temp_arr[i])
            ans = ans.next
        return ans_head
            
                    
                    
                    
        